moz_security/sandboxes/heka/analysis/moz_security_cloudtrail.lua

```diff
@@ -0,0 +1,147 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Mozilla Security Cloudtrail notifications
+
+## Sample Configuration
+```lua
+filename = "moz_security_cloudtrail.lua"
+message_matcher =
+ticker_interval = 0
+process_message_inject_limit = 1
+
+-- cloudtrail events to alert on (required)
+events = {
+    {
+        description = "mfa disabled",
+        fields = {
+            { "eventName", "^DeleteVirtualMFADevice$" },
+        }
+    },
+    {
+        description = "mfa disabled",
+        fields = {
+            { "eventName", "^DeactivateMFADevice$" },
+        }
+    },
+    {
+        description = "access key created",
+        fields = {
+            { "eventName", "^CreateAccessKey$" }
+        }
+    },
+    {
+        description = "IAM action in production account from console without mfa",
+        fields = {
+            { "eventSource", "^iam.amazonaws.com$" },
+            { "recipientAccountId", "^1234567890$" },
+            { "userIdentity.invokedBy", "^signin.amazonaws.com$" },
+            { "userIdentity.sessionContext.attributes.mfaAuthenticated", "^false$" }
+        }
+    }
+}
+
+-- mapping of aws account ids to human-friendly names (optional)
+aws_account_mapping = {
+    {"5555555555", "dev"},
+    {"1234567890", "prod"}
+}
+
+
+-- module makes use of alert output and needs a valid alert configuration
+alert = {
+    modules = { }
+}
+```
+--]]
+
+require "cjson"
+
+local alert = require "heka.alert"
+
+local cfevents = read_config("events")
+local cfaccount_name_mapping = read_config("aws_account_mapping")
+
+local payloadfields = {
```

Comment:
This seems tedious, we could just pull the message down in one shot and not have to request it one field at a time.  However, Fields will be in the native array format http://mozilla-services.github.io/lua_sandbox/heka/message.html

Reply:
:+1: totally agree, but looking at the [`read_message`](https://mozilla-services.github.io/lua_sandbox/heka/analysis.html#read_message) docs, it looks like you cannot do something like `read_message("Fields")`? Am I missing something or is there another way to do this?

Reply:
```lua
local msg = decode_message(read_message("raw"))
local j = cjson.encode(msg.Fields) -- however, this will give you the array schema, not the hash
```